html2/tree_builder.cpp

```diff
@@ -0,0 +1,229 @@
+#include "html2/tree_builder.h"
+
+#include "dom2/document_type.h"
+
+#include <spdlog/spdlog.h>
+
+#include <exception>
+#include <functional>
+
+using namespace std::literals;
```

Comment:
Did not see anything that was used from this namespace. But I might have missed something :)

Reply:
No, you're right. :D I probably did use it at some point, but I removed a lot of wip things when cleaning this before pushing it. Good catch!